File: frontend/src/api/listener.ts
import request from './request'

export interface Listener {
  id: number
  name: string
  host: string
  port: number
  protocol: string
  status: string
  connections: number
}

export const listenerApi = {
  list(): Promise<Listener[]> {
    return request.get('/listeners')
  },
  create(data: { name: string; host: string; port: number; protocol?: string }): Promise<Listener> {
    return request.post('/listeners', data)
  },
  start(id: number): Promise<void> {
    return request.post(`/listeners/${id}/start`)
  },
  stop(id: number): Promise<void> {
    return request.post(`/listeners/${id}/stop`)
  },
  delete(id: number): Promise<void> {
    return request.delete(`/listeners/${id}`)
  },
}

File: internal/model/listener.go
package model

// Listener 反向 Shell 监听器
type Listener struct {
	BaseModel
	Name        string `gorm:"size:128;not null" json:"name"`
	Host        string `gorm:"size:64;not null" json:"host"`
	Port        int    `gorm:"not null" json:"port"`
	Protocol    string `gorm:"size:16" json:"protocol"` // tcp / udp
	Status      string `gorm:"size:16" json:"status"`   // running / stopped
	Connections int    `json:"connections"`
	UserID      uint   `gorm:"index" json:"user_id"`
}

func (Listener) TableName() string {
	return "listeners"
}
